### PR TITLE
fix problem with thread.y and z index being incorrect in some hardware

### PR DESCRIPTION
**Pull request by nickhingston**
Hi, 

Managed to find a couple of hours to look at this issue...

please find a fix for #241 and #251 

Test case for this issue added too - but i don't seem to be able to run the full suite! If I comment out all other tests bar mine its ok...but this kinda defeats the point of a test suite!

I get "Too many active WebGL contexts. Oldest context will be lost." - if one of these has happened, my (and some others) will fail...

New issue?! (it's broken for me without my changes)

### CODE REVIEW

File: test/all.html [hunk 1]
```diff
@@ -58,6 +58,7 @@
   <script src="issues/207-same-function-reuse.js"></script>
 	<script src="issues/212-funky-function-support.js"></script>
   <script src="issues/233-kernel-map-float-output.js"></script>
+  <script src="issues/241-CPU-vs-GPU-maps-output-differently.js"></script>
```
Comment (robertleeplummerjr) [inline test/all.html]: Can you push up your test?

**Comment by robertleeplummerjr** [comment]
Two requests:
* Can you wrap the test in a `(function() { [test here] })();` like the other tests?  When I run the test, `input` gets overwritten and the test sometimes fails.
* Can you use `var` in tests?  We try and have tests pass in older versions of browsers (like modern safari), so we can continue to test in the duration we need to?

Fantastic work!

**Comment by nickhingston** [comment]
sure, you want me to fix 259-atan2.js too, which is where i templated it from?!

i guess there are others 263- etc...

seems there are other es6 features in some of these tests too...  e.g. () => {

Looks like you need a separate task to sort out the test framework a bit, but will fix this file as its new.

**Comment by robertleeplummerjr** [comment]
@nickhingston, fantastic job!  I look forward to more!

**Pull request merged**